docs/react-v9/contributing/rfcs/shared/build-system/stop-styles-transforms.md

```diff
@@ -0,0 +1,162 @@
+  -->
+
+<div class="main-order0 main-order1"></div>
+```
+
+## Problem statement
+
+Integrating prefixing into Griffel is relatively straightforward. However, we currently undergo a pre-processing step in `@fluentui/react-components` that modifies JavaScript code.
+
+<details>
+  <summary>Original code</summary>
+
+```ts
+// packages/react-components/react-menu/src/components/MenuDivider/useMenuDividerStyles.styles.ts
+// 📝 output is simpfied
+
+const useStyles = makeStyles({
+  root: {
+    ...shorthands.margin('4px', '-5px', '4px', '-5px'),
+  },
+});
+```
+
+</details>
+
+<details>
+  <summary>Pre-processed code</summary>
+
+```ts
+// @fluentui/react-menu/lib/components/MenuDivider/useMenuDividerStyles.styles.js
+// 📝 output is simpfied
+
+const useStyles = /*#__PURE__*/ __styles(
+  {
+    root: {
+      B6of3ja: 'fvjh0tl',
+      t21cq0: ['f1rnx978', 'f1q7jvqi'],
+    },
+  },
+  {
+    d: ['.fvjh0tl{margin-top:4px;}', '.f1rnx978{margin-right:-5px;}', '.f1q7jvqi{margin-left:-5px;}'],
+  },
+);
+```
+
+</details>
+
+Addressing prefixing for pre-processed styles presents a challenge, as they have already been transformed into CSS rules and classes.
+
+## Proposal
+
+The long-term solution is to discontinue the pre-processing of styles in `@fluentui/react-components` and transfer this responsibility to the consuming application.
+
+> Note: Applications already undertake this responsibility, as first and third-party packages do not include pre-processed styles.
+
+### Option A: Stop pre-processing styles
+
+This option involves eliminating the pre-processing step from `@fluentui/react-components`, shifting the responsibility to the consuming application.
+
+This is the simplest option, albeit it necessitates adjustments in the consuming app and is not backward compatible. Consequently, apps not utilizing [AOT in Griffel](https://griffel.js.org/react/ahead-of-time-compilation/introduction) will experience noticeable slowdowns during initial loading.
+
+## Pros and Cons
+
+- 👍 Simple & easy
+- 👍 Not backward compatible
+
+### Option B: Ship ESM output with unprocessed styles
+
+This option is more complex to maintain but ensures backward compatibility and avoids breaking changes.
+
+We will offer three outputs for each package under the following `exports` in `package.json`:
+
+- `"import": "./lib/index.js"` (as is) - ESM output with pre-processed styles (default for bundlers)
+- `"require": "./lib-commonjs/index.js"` (as is) - CJS output with pre-processed styles (default for Node.js)
+- `"import-raw": "./lib-raw/index.js"` (🆕)- ESM output with unprocessed styles
+
+Consumers desiring to use the class prefixing feature will need to configure their bundler to use the `"import-raw"` output. For example, in Webpack:
+
+```js
+// webpack.config.js
+
+module.exports = {
+  resolve: {
+    conditionNames: ['import-raw', '...'],
+  },
+};
+```
+
+- 👍 Backward compatible
+- 👍 No changes required in consuming apps
```

Comment:
there actually  will be changes required in user-land. in order to use griffel prefixing they would need to point to this new export map. am I missing something ?

Reply:
Both solutions will require changes in userland. For smaller apps styling conflicts might not be a problem. Larger apps should be affected since there is a high likelihood of either
* pulling in a partner bundle
* pulling in a partner package that uses its own renderer

Reply:
> Both solutions will require changes i

which "both" solutions are you refering to ?

with this new export maps proposal:

- if folks have no issues they don't need to change anything
- if folks have issues they need to change webpack configs to point to different export map 


am I missing something ?